src/main/kotlin/parts/wisdom/arcraider/visualization/GridPane.kt

```diff
@@ -8,12 +8,18 @@ import javax.swing.JPanel
 // we don't fill right to the edge of the screen.
 private const val PANEL_INSET_PIXELS = 10
 
-class GridPane(val grid: Grid) : JPanel() {
+class GridPane(var grid: Grid) : JPanel() {
 
-    private val xPixelsPerSquare  by lazy { (width - PANEL_INSET_PIXELS) /  grid.widthSquares }
-    private val yPixelsPerSquare by lazy { (height - PANEL_INSET_PIXELS) / grid.heightSquares }
-    private val xStartPixels by lazy { (width - xPixelsPerSquare * grid.widthSquares) / 2 }
-    private val yStartPixels by lazy { (height - yPixelsPerSquare * grid.heightSquares) / 2 }
+    fun resetGrid(grid: Grid) {
+        this.grid = grid
+        this.revalidate()
+        this.repaint()
+    }
+
+    fun xPixelsPerSquare() : Int = (width - PANEL_INSET_PIXELS) /  grid.widthSquares
+    fun yPixelsPerSquare() : Int = (height - PANEL_INSET_PIXELS) / grid.heightSquares
+    fun xStartPixels() : Int = (width - xPixelsPerSquare() * grid.widthSquares) / 2
+    fun yStartPixels() : Int = (height - yPixelsPerSquare() * grid.heightSquares) / 2
```

Comment:
these can be private

Reply:
Would it be more natural to make them computed properties?

Reply:
I played around with that but they need to be computed again every time the grid is updated. 

You can make a function like computePixels() that gets called on reset that runs all this code, but then it also needs to be called on initialization, and only after the height / width of the pane has been computed ie kotlin's init block is too soon.  Something like kotlin's lazy would be good for this except lazy only works easily for val types and these properties would need to be var to update the values on reset. 

So the simplest code I could find  to easily recompute all of these was these 4 functions.